packages/api/src/graphql/mutation.private.ts

```diff
@@ -183,12 +192,8 @@ export const GraphQLAdminMutation = new GraphQLObjectType<undefined, Context>({
     updatePeerProfile: {
       type: GraphQLNonNull(GraphQLPeerProfile),
       args: {input: {type: GraphQLNonNull(GraphQLPeerProfileInput)}},
-      async resolve(root, {input}, {hostURL, authenticate, dbAdapter}) {
-        const {roles} = authenticate()
-        authorise(CanUpdatePeerProfile, roles)
-
-        return {...(await dbAdapter.peer.updatePeerProfile(input)), hostURL}
-      }
+      resolve: (root, {input}, {hostURL, authenticate, prisma: {peerProfile}}) =>
+        upsertPeerProfile(input, hostURL, authenticate, peerProfile)
```

Comment:
In my opinion the authentication and authorization should be handled right here. Why?
- Separation of concerns: Authentication and authorization is much closer to GraphQL than inside business logic and database requests, like it is now.
- All authorization and authentication is done in one file and in one place which is mutation.private.ts. This makes security concerns much easier, less complex and less distributed.
- We don't have to pass authentication context like upsertPeerProfile(input, hostURL, **authenticate**, peerProfile)

Reply:
Hey hey, the idea of these functions is so that the mutation/query files aren't huge as they are right now. The function in itself doesn't really authenticate and imo it's a very misleading name.

```typescript
  authenticate() {
      if (!session) {
        throw new AuthenticationError('Invalid session!')
      }

      if (!isSessionValid) {
        throw new TokenExpiredError()
      }

      return session
    }
```

I agree with you on principle (but not implementation) as I think this should be abstracted away and shouldn't have to be repeated in every single method.

Is this blocking for you?

Reply:
Created a Jira story

Reply:
Amazing, thank you!